bytestring.cabal

```diff
@@ -113,10 +113,19 @@ library
                     -fmax-simplifier-iterations=10
                     -fdicts-cheap
                     -fspec-constr-count=6
+  
+  c-sources:        cbits/fpstring.c
+                    cbits/itoa.c
+  if (arch(x86_64) || arch(i386))
+    c-sources:        cbits/x86/is-valid-utf8.c
+  else
+    c-sources:        cbits/is-valid-utf8.c
 
-  c-sources:         cbits/fpstring.c
-                     cbits/itoa.c
   cc-options:        -std=c11
+  
+  if os(windows)
+    extra-libraries:  gcc_s
```

Comment:
As suggested at https://gitlab.haskell.org/ghc/ghc/-/issues/20525#note_385580, could you please try adding `gcc` as well?